### PR TITLE
docs(gatsby): stub out docs for source plugin authors implementing support for Cloud Content Loader

### PR DESCRIPTION
**Pull request by TylerBarnes**

### CODE REVIEW

File: docs/docs/creating-a-source-plugin.md [hunk 1]
```diff
@@ -122,6 +122,82 @@ This can reduce the time it takes repeated data fetching operations to run if yo
 
 You can read more about the cache API, other types of plugins that leverage the cache, and example open source plugins that use the cache in the [build caching guide](/docs/build-caching).
 
+### Adding support for Gatsby Cloud Content Loader
+
+Content Loader is a service that will allow your CMS backend to route from a piece of content to a finally built page in Gatsby. Currently, the primary use-case for this is implementing Content Previews. This allows site admins to preview their content in a Gatsby Preview instance before they decide to push it to their production site.
```
Comment (DanielSLew) [inline docs/docs/creating-a-source-plugin.md:127]:
```suggestion
Content Loader is a service that will allow your CMS backend to route from a piece of content to a finished page on your Gatsby site where that content lives. Currently, the primary use-case for this is implementing Content Previews. This allows site admins to preview their content in a Gatsby Preview instance before they decide to push it to their production site.
```

File: docs/docs/creating-a-source-plugin.md [hunk 2]
```diff
@@ -122,6 +122,82 @@ This can reduce the time it takes repeated data fetching operations to run if yo
...
+
+#### How Content Loader works for users
+
+The feature will display a loading page while the users content is being built in Gatsby. Once the build is finished, the loader will route the user to the correct page. If any errors occur, the loader will let the user know. This way site admins wont be left guessing when they can view their preview content.
```
Comment (DanielSLew) [inline docs/docs/creating-a-source-plugin.md:131]:
```suggestion
This feature will display a loading page while the users content is being built in Gatsby Cloud. Once the build is finished, the loader will route the user to the correct page. If any errors occur, the loader will let the user know. This takes care of the awkward moment where you're trying to preview content that hasn't been built yet and site admins won't be left guessing when they can view their preview content.
```

File: docs/docs/creating-a-source-plugin.md [hunk 3]
```diff
@@ -122,6 +122,82 @@ This can reduce the time it takes repeated data fetching operations to run if yo
...
+
+TODO: write and then link to the site developer docs on how ownerNodeId in createPage relates to this feature.
+
+TODO: what is a node manifest?
+
+#### Why use Content Loader
+
+Using Content Loader has some advantages over having your CMS load and route to previews.
+
+1. The loader is aware of wether or not the build has finished, allowing the user to be routed to the right page at the right time.
```
Comment (DanielSLew) [inline docs/docs/creating-a-source-plugin.md:141]:
```suggestion
1. The loader is aware of whether or not the build has finished, allowing the user to be routed immediately when it's ready.
```

We mention being routed to the right page on the third point, so we can take it out of this point.

Comment (TylerBarnes) [reply docs/docs/creating-a-source-plugin.md:141]: I think it's ok to mention it in both places here 👌

**Comment by TylerBarnes** [comment]
@DanielSLew I committed your suggestions but I changed them slightly. The audience for this part of the docs will be source plugin authors rather than Gatsby users, so the Gatsby sites using the loader wont be theirs.

**Pull request closed without merging**